token/sdk/tms/db/tms.go

```diff
@@ -134,28 +118,16 @@ func (p *PostInitializer) ConnectNetwork(networkID, channel, namespace string) e
 		Channel:   channel,
 		Namespace: namespace,
 	}
-	notifier, err := events.GetPublisher(p.sp)
-	if err != nil {
-		return errors.WithMessagef(err, "failed to get event publisher")
-	}
-	tokenDB, err := tokendb.GetByTMSId(p.sp, tmsID)
-	if err != nil {
-		return errors.WithMessagef(err, "failed to get token db")
-	}
-	tokenStore, err := db.NewTokenStore(notifier, tokenDB, tmsID)
-	if err != nil {
-		return errors.WithMessagef(err, "failed to get token store")
-	}
 	if err := n.ProcessorManager().AddProcessor(
 		namespace,
 		fabric2.NewTokenRWSetProcessor(
 			n.Name(),
 			namespace,
+			func() *tokens2.Tokens {
```

Comment:
I may be missing it, but if there is no other way of getting the tokens, then it would make more sense to pass the service directly

Reply:
I just want to postone this call to when it will be really needed so to reduce resource allocation.
Wdyt?

Reply:
Then you will call this method every time you need the tokens. We could use some lazy initialisation to combine th best of both worlds. We could have a generic solution like
```
type lazyGetter[V] struct {
v V
provider func() V
once sync.Once 
}

func NewLazyGetter[V](provider func() V) *lazyGetter[V] {
 return &lazyGetter[V]{provider: provider}
}

func (g *lazyGetter[V]) Get() V {
 g.once.Do(func() {
  g.v = g.provider()
 })
 return g.v
}
```

Reply:
👍